contracts/nexus-gateway/src/contract/migrations/v1_0_0.rs

```diff
@@ -0,0 +1,126 @@
+use axelar_wasm_std::address;
+use axelar_wasm_std::error::ContractError;
+use cosmwasm_schema::cw_serde;
+use cosmwasm_std::{Addr, DepsMut};
+use cw_storage_plus::Item;
+
+use crate::contract::{CONTRACT_NAME, CONTRACT_VERSION};
+use crate::msg::MigrateMsg;
+use crate::state;
+
+const BASE_VERSION: &str = "1.0.0";
+const CONFIG: Item<Config> = Item::new("config");
+
+pub fn migrate(deps: &mut DepsMut, msg: MigrateMsg) -> Result<(), ContractError> {
+    cw2::assert_contract_version(deps.storage, CONTRACT_NAME, BASE_VERSION)?;
+
+    let axelarnet_gateway = address::validate_cosmwasm_address(deps.api, &msg.axelarnet_gateway)?;
+    migrate_config(deps, axelarnet_gateway)?;
+
+    cw2::set_contract_version(deps.storage, CONTRACT_NAME, CONTRACT_VERSION)?;
+
+    Ok(())
+}
+
+fn migrate_config(deps: &mut DepsMut, axelarnet_gateway: Addr) -> Result<(), ContractError> {
+    let config = CONFIG.load(deps.storage)?;
+    CONFIG.remove(deps.storage);
+
+    state::save_config(
+        deps.storage,
+        state::Config {
+            nexus: config.nexus,
+            router: config.router,
+            axelarnet_gateway,
+        },
+    )
+    .map_err(Into::into)
+}
+
+#[cw_serde]
```

Comment:
Can we mark this as deprecated?
```suggestion
#[cw_serde]
#[deprecated(since = "1.0.0", note = "only used during migration")]
```